velox/connectors/hive/SplitReader.cpp

```diff
@@ -310,15 +310,11 @@ void SplitReader::setRowIndexColumn(
   auto rowIndexColumnName = rowIndexColumn->name();
   auto rowIndexMetaColIdx =
       readerOutputType_->getChildIdxIfExists(rowIndexColumnName);
-  if (rowIndexMetaColIdx.has_value() &&
-      !fileType->containsChild(rowIndexColumnName) &&
```

Comment:
So we no longer keep the column with same name in file?

Reply:
in vinalia spark that's true,  if a column been marked to row index column it will always be treat as row index column dont care about if the file has same name data column.  The change here is make sure if engine part set the column is to `kRowindex` type column, then should always treat it `RowIndex` column not data column, otherwise the behavior is strange and inconsistent.

If the file has same name column, and engine want to let the `data` column has high priority compare with same name row index column, then engine has responsibility to not set `kRowIndex` handle for it.